src/main/java/org/openrewrite/java/testing/assertj/IsEqualToNull.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2024 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.testing.assertj;
+
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Preconditions;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.MethodMatcher;
+import org.openrewrite.java.search.UsesMethod;
+import org.openrewrite.java.tree.Expression;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.J.MethodInvocation;
+import org.openrewrite.java.tree.JavaType.Method;
+
+import java.util.Collections;
+
+/**
+ * AssertJ has a more idiomatic way of asserting that an object is null. This
+ * recipe will find instances of:
+ * <p>
+ * -`assertThat(Object).isEqualTo(null)` and replace them with `isNull()`.
+ */
+public class IsEqualToNull extends Recipe {
+
+    private static final MethodMatcher IS_EQUAL_TO = new MethodMatcher("org.assertj.core.api.AbstractAssert isEqualTo(..)");
```

Comment:
So, is there any way I can match all subclasses of AbstractAssert?

for instance this recipe is valid for Strings too
```
org.assertj.core.api.AbstractStringAssert isEqualTo(java.lang.String)
```


But I have no idea what all permutations of `isEqualTo` exist out there.

So, is there any way I can catch them all?!

Reply:
Yes absolutely! The method matcher has an optional second argument boolean to match overrides.
![image](https://github.com/user-attachments/assets/94468624-f6e4-4895-8888-59cbf0d0b10d)

I've applied that and a small change to switch to `Traits.methodAccess` already.